### PR TITLE
Allow for domain-wide delegation in google cloud apps

### PR DESCRIPTION
**Pull request by mtagle**
add a param on google cloud app hooks and operators to allow passing an account to impersonate.

some documentation from google on the subject: https://developers.google.com/identity/protocols/OAuth2ServiceAccount?hl=en_US

### CODE REVIEW

**Comment by mtagle** [comment]
So it looks like this is only failing due to coveralls. Any chance you could overlook that and merge it anyway? :)

File: airflow/contrib/hooks/bigquery_hook.py [hunk 1]
```diff
@@ -27,12 +27,15 @@ class BigQueryHook(GoogleCloudBaseHook):
     """
     conn_name_attr = 'bigquery_conn_id'
 
-    def __init__(self, scope='https://www.googleapis.com/auth/bigquery', bigquery_conn_id='bigquery_default'):
+    def __init__(self,
+                 scope='https://www.googleapis.com/auth/bigquery',
+                 bigquery_conn_id='bigquery_default',
+                 sub=None):
```
Comment (criccomini) [inline airflow/contrib/hooks/bigquery_hook.py]: Slightly more descriptive name would be nice (delegate_to?)

File: airflow/contrib/hooks/bigquery_hook.py [hunk 2]
```diff
@@ -27,12 +27,15 @@ class BigQueryHook(GoogleCloudBaseHook):
...
         """
         :param scope: The scope of the hook.
         :type scope: string
```
Comment (criccomini) [inline airflow/contrib/hooks/bigquery_hook.py]: PyDoc for param.

File: airflow/contrib/hooks/bigquery_hook.py [hunk 3]
```diff
@@ -27,12 +27,15 @@ class BigQueryHook(GoogleCloudBaseHook):
...
         """
-        super(BigQueryHook, self).__init__(scope, bigquery_conn_id)
+        super(BigQueryHook, self).__init__(scope, bigquery_conn_id, sub)
```
Comment (criccomini) [inline airflow/contrib/hooks/bigquery_hook.py]: PyDoc for params above.

File: airflow/contrib/hooks/gc_base_hook.py [hunk 4]
```diff
@@ -36,6 +39,10 @@ def _authorize(self):
         service_account = connection_extras.get('service_account', False)
         key_path = connection_extras.get('key_path', False)
 
+        kwargs = {}
+        if self.sub:
+            kwargs['sub'] = self.sub
```
Comment (criccomini) [inline airflow/contrib/hooks/gc_base_hook.py]: I think you can do this more idiomatically as:

```
kwargs = {'sub': self.sub} if self.sub else {}
```

**Comment by mistercrunch** [comment]
It's failing because of the test are failing, I think you most likely forked at a bad time. Can you please rebase your branch, that will re-trigger the tests.

File: airflow/contrib/hooks/gc_base_hook.py (hunk 4 above)
Comment (mtagle) [inline airflow/contrib/hooks/gc_base_hook.py]: my thought is this is a bit more extensible if we ever need to add anything else to kwargs.

**Comment by mistercrunch** [comment]
@criccomini , do I have your blessing to merge this?

**Comment by criccomini** [comment]
+1 on my end

**Comment by mistercrunch** [comment]
LGTM, thanks for your contribution!

**Pull request merged**